inst/stan/epinowcast.stan

```diff
@@ -428,31 +438,45 @@ generated quantities {
     if (model_miss) {
       // Allocate proportion that are not reported
       log_exp_miss_by_ref = apply_missing_reference_effects(
-        1, s, log_exp_obs, sdmax, csdmax, miss_ref_lprop
+        1, s, log_exp_obs_all, sdmax, csdmax, miss_ref_lprop
       );
       log_exp_miss_ref = log_expected_by_report(
         log_exp_miss_by_ref, obs_by_report
       );
       // Allocate proportion that are reported
-      log_exp_obs = apply_missing_reference_effects(
-        1, s, log_exp_obs, sdmax, csdmax, log1m_exp(miss_ref_lprop)
+      log_exp_obs_all = apply_missing_reference_effects(
+        1, s, log_exp_obs_all, sdmax, csdmax, log1m_exp(miss_ref_lprop)
       );
     }
+    // Allocate to just those actually observed
+    log_exp_obs = allocate_observed_obs(
+      1, s, log_exp_obs_all, sl, csl, sdmax, csdmax
+    );
     // Draw from observation model for observed counts with report and reference
-    pp_obs_tmp = obs_rng(log_exp_obs, phi, model_obs);
+    pp_obs_all = obs_rng(log_exp_obs_all, phi, model_obs);
+    // Allocate to just those actually observed
+    pp_obs_tmp = allocate_observed_obs(
+      1, s, pp_obs_all, sl, csl, sdmax, csdmax
+    );
+    // Allocate to just those in the likelihod
+    pp_obs_lik = allocate_observed_obs(
+      1, s, pp_obs_tmp, lsl, clsl, sl, csl
+    );
     } 
 
     // Likelihood by snapshot (rather than by observation)
     profile("generated_loglik") {
     if (ologlik) {
       for (i in 1:s) {
-        array[3] int l = filt_obs_indexes(i, i, csl, sl);
-        array[3] int f = filt_obs_indexes(i, i, csdmax, sdmax);
+        array[3] int l = filt_obs_indexes(i, i, cnsl, nsl);
         log_lik[i] = 0;
-        for (j in 1:sl[i]) {
-          log_lik[i] += obs_lpmf(
-            flat_obs[l[1] + j - 1]  | log_exp_obs[f[1] + j - 1], phi, model_obs
-          );
+        if (nsl[i]) {
+          for (j in 1:nsl[i]) {
+            log_lik[i] += obs_lpmf(
```

Comment:
Need to correct this so that obs are properly linked to expected values. When doing so should generalise to have a log-likelihood value per observation vs per snapshot.

Reply:
resolved in [6c4f127](https://github.com/epinowcast/epinowcast/pull/327/commits/6c4f12767d7ba4d3b226bc6d019c9ddee50bd4d6)